zebra/zebra_rib.c

```diff
@@ -4373,6 +4382,21 @@ int rib_add_multipath(afi_t afi, safi_t safi, struct prefix *p,
 		nhe.id = re->nhe_id;
 
 	n = zebra_nhe_copy(&nhe, 0);
+
+	if (re->type == ZEBRA_ROUTE_KERNEL) {
+		struct interface *ifp;
+
+		if (p->family == AF_INET6 &&
+		    IN6_IS_ADDR_LINKLOCAL(&p->u.prefix6)) {
```

Comment:
interesting - don't protocols sometimes want to know about the link-locals though?

Reply:
not as a kernel route.  We already have code that handles the link locals.  This is just a special case for how we are receiving the v6 LL routes from the kernel as far as I can tell.